packages/core/useScriptTag/index.ts

```diff
@@ -0,0 +1,142 @@
+import { isString, MaybeRef, tryOnMounted, tryOnUnmounted } from '@vueuse/shared'
+import { ref } from 'vue-demi'
+import { ConfigurableDocument, defaultDocument } from '../_configurable'
+
+export interface UseScriptTagOptions extends ConfigurableDocument {
+  immediate?: boolean
+  async?: boolean
+  type?: string
+  crossOrigin?: 'anonymous' | 'use-credentials'
+  referrerPolicy?: 'no-referrer' | 'no-referrer-when-downgrade' | 'origin' | 'origin-when-cross-origin' | 'same-origin' | 'strict-origin' | 'strict-origin-when-cross-origin' | 'unsafe-url'
+  noModule?: boolean
+  defer?: boolean
+}
+
+/**
+ * Async script tag loading.
+ *
+ * @see   {@link https://vueuse.js.org/useScriptTag}
+ * @param src
+ */
+export function useScriptTag(
+  src: MaybeRef<string>,
+  onLoaded: Function = () => {},
+  {
+    immediate = true,
+    async = true,
+    document = defaultDocument,
+    type = 'text/javascript',
+    crossOrigin,
+    referrerPolicy,
+    noModule,
+    defer,
+  }: UseScriptTagOptions = {},
+) {
+  const scriptTag = ref<HTMLScriptElement>()
+
+  /**
+   * Load the script specified via `src`.
+   *
+   * @param waitForScriptLoad Whether if the Promise should resolve once the "load" event is emitted by the <script> attribute, or right after appending it to the DOM.
+   * @returns Promise<HTMLScriptElement>
+   */
+  const load = (waitForScriptLoad = true): Promise<HTMLScriptElement|boolean> =>
```

Comment:
We should have an internal promise instance here (singleton) to prevent calling load() twice.

Reply:
I suggest you to review my updates in that commit: https://github.com/antfu/vueuse/pull/274/commits/fdee84562eb65bd20294d2d918501368be69615c

Thanks again for your review.